pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java

```diff
@@ -194,8 +194,7 @@ public BigDecimal getBigDecimal(int columnIndex, int scale)
       throws SQLException {
     try {
       String value = getString(columnIndex);
-      BigDecimal bigDecimal = new BigDecimal(value).setScale(scale);
-      return bigDecimal;
+      return wasNull() ? null : new BigDecimal(value).setScale(scale);
```

Comment:
Seems `getString()` already checks whether the value is null, so here we can simply check if the `value == null`. Same for other places

Reply:
Changed.